deploy/kubedirector/crd-settings.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: kubedirectorsettingss.kubedirector.bluedata.io
+spec:
+  group: kubedirector.bluedata.io
+  names:
+    kind: KubeDirectorSettings
+    listKind: KubeDirectorSettingsList
+    plural: kubedirectorsettingss
+    singular: kubedirectorsettings
+  scope: Namespaced
+  version: v1alpha1
+  validation:
+    openAPIV3Schema:
+      properties:
```

Comment:
I think you could get k8s to do the name enforcement by adding this to the schema:
```yaml
        metadata:
          properties:
            name:
              type: string
              pattern: '^kd-settings$'
```

Although:

If we're thinking of allowing per-managed-namespace overrides of (at least some of) these settings in the future -- and also, just kind of thinking/commenting on the basic idea of using a CR for this purpose...

Maybe we should change the allowed name from "kd-settings" to "kd-global-settings". Then the future per-namespace CRs could use a name like "kd-namespace-settings". This would allow for example the case of KD managing clusters in its own namespace without having a collision between the two kinds of settings CRs.

It's also a little less odd to have a new CRD if we do imagine possibly having multiple CRs in the future (as opposed to forever just a single CR using that CRD). That does kind of raise the question of whether per-namespace settings will have the same schema as global settings, but we can fight that fight when we get to it.

If we think that's our path though, then we can:

* Change the name to "kd-global-settings".
* We will probably want to have the validation handler reject any kd-global-settings CR created in a watched namespace that is not the same as KD's own namespace, just to avoid confusion. That's a moot point at the moment though since we are just having KD watch its own namespace.
* We can enforce the kd-global-settings name in the schema here (I think) with the pattern `'^kd-global-settings$'` and not need to enforce it in the validation handler.
* If/when we introduce per-namespace settings, the schema pattern will need to be changed to allow both names, i.e. `'^kd-global-settings$|^kd-namespace-settings$'`

Reply:
Changed it to kd-global-settings